### PR TITLE
Make watched_files_digest thread safe

### PR DESCRIPTION
**Pull request by justin808**
Based on PR https://github.com/rails/webpacker/pull/3218

I detected the issue with #3218 when I noticed the timing was impossibly fast

Note, this method is still crazy slow on big projects. 

On a big project:
       user     system      total        real
   1.985895   2.741152   4.727047 (  9.214857)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,12 @@
 
 Please see [UPGRADE GUIDE](./docs/v6_upgrade.md) for more information.
 
-- `node_modules` will no longer be compiled by default. This primarily fixes [rails issue #35501](https://github.com/rails/rails/issues/35501) as well as [numerous other webpacker issues](https://github.com/rails/webpacker/issues/2131#issuecomment-581618497). The disabled loader can still be required explicitly via:
+- `node_modules` will no longer be babel transfomed compiled by default. This primarily fixes [rails issue #35501](https://github.com/rails/rails/issues/35501) as well as [numerous other webpacker issues](https://github.com/rails/webpacker/issues/2131#issuecomment-581618497). The disabled loader can still be required explicitly via:
```
Comment (guillaumebriday) [inline CHANGELOG.md:9]:
```suggestion
- `node_modules` will no longer be babel transformed compiled by default. This primarily fixes [rails issue #35501](https://github.com/rails/rails/issues/35501) as well as [numerous other webpacker issues](https://github.com/rails/webpacker/issues/2131#issuecomment-581618497). The disabled loader can still be required explicitly via:
```

**Comment by justin808** [comment]
We should run the docs through Grammarly! Thanks for the typo catch, @guillaumebriday!

**Comment by justin808** [comment]
Thanks, @nguyenthetoan. I applied your changes.

**Comment by justin808** [comment]
Merged in https://github.com/shakacode/webpackr/pull/6.

**Pull request closed without merging**